### DISCUSSION TITLE
feat: git push事件增加“只保留最新触发的任务”的开关

### DISCUSSION THREAD

**Discussion by zanyzhao** [Tencent/bk-ci#4251] [Ideas]
**What would you like to be added**:
当git事件触发选择push时，出现等待队列只保留最新触发的任务。
**Why is this needed**:
主干分支不断有提交，而私有构建资源不足以支撑，可以由用户选择性开启这个特性，类似 #411

**Comment by PandaraWen** [comment]
这边有类似的诉求。测试回归验证问题的时候我们会频繁在分支上提交代码，为了方便出包配置了 git push 触发自动打包。测试回归问题的时候需要取最新包来回归，等待队列中保留最新一次触发的任务可以节省资源。

**Comment by fatterx** [comment]
我们有代码merge到master后，自动发布到maven的需求，如果短期内频繁merge，会有排队的任务在发布，但实际上之前的任务即使发布成功也会被后面新的任务覆盖掉，浪费资源

**Comment by zanyzhao** [comment]
正在运行中的任务在新的push事件来了后，是直接终止，还是继续执行？
后续如果加了这个特性 #1156 ，可能会冲突
